component/src/test/java/org/wso2/carbon/identity/authenticator/bitly/BitlyTest.java

```diff
@@ -0,0 +1,115 @@
+import org.mockito.Mock;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import org.powermock.reflect.Whitebox;
+import org.testng.Assert;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+import org.wso2.carbon.identity.application.authentication.framework.model.AuthenticatedUser;
+import org.wso2.carbon.identity.application.authenticator.oidc.OIDCAuthenticatorConstants;
+import org.wso2.carbon.identity.application.common.model.ClaimMapping;
+import org.wso2.carbon.identity.application.common.util.IdentityApplicationConstants;
+
+@RunWith(PowerMockRunner.class)
+
+@PrepareForTest({AuthenticatedUser.class, OAuthAuthzResponse.class,
+        OAuthClientRequest.class, URL.class, OAuthClient.class})
+
+public class BitlyTest {
+
+    BitlyAuthenticator bitlyAuthenticator;
+
+    @Mock
+    private OAuthClientResponse oAuthClientResponse;
+
+    @Mock
+    private OAuthClient mockOAuthClient;
+
+    @Mock
+    private OAuthClientRequest mockOAuthClientRequest;
+
+    @DataProvider(name = "authenticatorProperties")
+    public Object[][] getAuthenticatorPropertiesData() {
+        Map<String, String> authenticatorProperties = new HashMap<>();
+        authenticatorProperties.put(OIDCAuthenticatorConstants.CLIENT_ID, "test-client-id");
+        authenticatorProperties.put(OIDCAuthenticatorConstants.CLIENT_SECRET, "test-client-secret");
+        authenticatorProperties.put("callbackUrl", "http://localhost:9443/commonauth");
+        authenticatorProperties.put("tokenEndpoint", "https://api-ssl.bitly.com/oauth/access_token");
+        authenticatorProperties.put("code", "dummy-code");
+        return new Object[][]{{authenticatorProperties}};
+    }
+
+    @BeforeMethod
+    public void setUp() {
+        bitlyAuthenticator = new BitlyAuthenticator();
+        initMocks(this);
+    }
+
+    @Test(description = "Test case for GetConfigurationProperties")
+    public void testGetConfigurationProperties() {
+        Assert.assertEquals(IdentityApplicationConstants.OAuth2.CALLBACK_URL,
+                bitlyAuthenticator.getConfigurationProperties().get(2).getName());
+        Assert.assertEquals(3, bitlyAuthenticator.getConfigurationProperties().size());
+    }
+
+    @Test(description = "Test case for getAccessRequest", dataProvider = "authenticatorProperties")
+    public void testGetAccessRequest(Map<String, String> authenticatorProperties) throws Exception {
+        OAuthClientRequest accessRequest = Whitebox.invokeMethod(bitlyAuthenticator, "getAccessRequest",
+                authenticatorProperties.get("tokenEndpoint"), authenticatorProperties.get(OIDCAuthenticatorConstants.CLIENT_ID),
+                authenticatorProperties.get("code"), authenticatorProperties.get(OIDCAuthenticatorConstants.CLIENT_SECRET),
+                authenticatorProperties.get("callbackUrl"));
```

Comment:
Use constants if possible for "tokenEndpoint", "code" and etc even though this  is a test case

Reply:
add constants in 2085640